
From: Jesse Barnes <jbarnes@engr.sgi.com>

sd->groups isn't initialized this early (I know because I ran into this
when I first tried to merge Suresh's patch too).  You've already got the
real fix in your tree, you just need this little bit torn out.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/kernel/sched.c |    2 --
 1 files changed, 2 deletions(-)

diff -puN kernel/sched.c~sched-merge-fix kernel/sched.c
--- 25/kernel/sched.c~sched-merge-fix	Wed Jul 28 15:03:12 2004
+++ 25-akpm/kernel/sched.c	Wed Jul 28 15:03:12 2004
@@ -3764,8 +3764,6 @@ __init static void arch_init_sched_domai
 		cpumask_t nodemask = node_to_cpumask(cpu_to_node(i));
 
 #ifdef CONFIG_NUMA
-		if (i != first_cpu(sd->groups->cpumask))
-			continue;
 		sd = &per_cpu(node_domains, i);
 		group = cpu_to_node_group(i);
 		*sd = SD_NODE_INIT;
_
